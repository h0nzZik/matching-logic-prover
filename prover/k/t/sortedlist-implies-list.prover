claim \implies( \and( sortedlist(variable("H"), variable("X"), variable("K"), variable("MAX"), .Patterns), .Patterns)
              , \and( list( variable("H"), variable("X"), variable("K"), .Patterns), .Patterns )
              )
strategy kt ; ( (right-unfold-Nth(0, 0) ; simplify ; direct-proof)
              | ( simplify ; direct-proof )
              | (right-unfold-Nth(0, 1) ; simplify ; direct-proof)
              )
