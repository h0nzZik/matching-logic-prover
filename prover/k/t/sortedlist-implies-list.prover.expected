<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( sortedlist ( variable ( "H" ) , variable ( "X" ) , variable ( "K" ) , variable ( "MAX" ) , .Patterns ) , .Patterns ) , \and ( list ( variable ( "H" ) , variable ( "X" ) , variable ( "K" ) , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
